3750/2/contexts/photo_check.rb

```diff
@@ -0,0 +1,34 @@
+module Contexts
+  module PhotoCheck
+    def photo_check(*)
+      proceed_to_geo_check && return if photo?
+      notify_with_reference(:photo_check_failure)
+      save_context(:photo_check)
+    end
+
+    private
+
+    def proceed_to_geo_check
+      photo_save
+      notify(:photo_check_success)
+      save_context(:geo_check)
+    end
+
+    def photo?
+      payload['photo']
+    end
+
```

Comment:
create interface like:
```
  PathGenerator.save_photo(session, payload['photo'])
```

move create_path photo_save to PathGenerator

do the same for geo_save
```
  PathGenerator.save_location(session, payload['location'])
```

rename PathGenerator to Storage

Reply:
```
storage = Storage.new(session, payload)
storage.save_location

Storage.save_location(session, payload)
```